src/extension.ts

```diff
@@ -66,14 +68,27 @@ interface LogChannel {
   hasMoreLogs?: boolean;
 }
 
+const loadEnvironments = async (
+  environmentsDataProvider: Env0EnvironmentsProvider,
+  environmentsTree: vscode.TreeView<Environment>
+) => {
+  try {
+    const { currentBranch } = getGitData();
+    environmentsTree.message = `loading environments from branch ${currentBranch}…`;
```

Comment:
can you add a spinner? from antd icons

Reply:
Do you think that testing here should be easy?
wanna give it a short try?

Reply:
About the spinner: i cannot add a spinner in this message since this message is a simple string.
i can try to do some trick with webview here while loading environments let me know if tis something that you want me to check.
i can also try to use TreeItem instead of view.message to show loading message but it can be weird since tree items are selectable  and also i dont think we can show spinner in tree items.
about the test: are you taking about integration tests with @vscode/test-electron?

Reply:
i will try to add the vs code withProgress indicator

Reply:
you mean the clock icon? yeah sounds good.

about electorn, I saw an example, and yeah I think it could be test using that.

Reply:
unit tests 
102d384f0a7cd40527c389ff47b4ab090e2a6f3b